daviaugustos/chat-app
# TL-04: The user should be able to double tap on a message to send a “Like” reaction.

# Title

[TL-04](https://trashlab.notion.site/Mobile-Take-Home-Assignment-10bdb96ac1e78033af18cdfc06a04fee): The user should be able to double tap on a message to send a “Like” reaction.

## Description

🏗️ Reorganized files to align with established pattern.
🔄 Replaced initial state mock structure with Zustand for state management.
👍 Introduced message liking functionality.
✌️ Implemented double-tap gesture for liking messages.
✨ Added animation and vibration feedback for like action.

## Screenshots

<img width="461" alt="image" src="https://github.com/user-attachments/assets/82d4a3a8-e837-401d-acb3-eef4144e5066">

## Testing

- Open the app;
- Click on a contact to open the respective chat message history.
- Type a message and hit the save button;
- Check if the message sent is added to the chat history.
- Hit double tap to like the message.
